pkg/archive/archive.go

```diff
@@ -83,12 +83,7 @@ func GenerateTarWithWriter(genFn func(TarWriter) error, twf TarWriterFactory) io
 		errChan <- closeErr
 	}()
 
-	closed := false
 	return ioutils.NewReadCloserWrapper(pr, func() error {
-		if closed {
-			return errors.New("reader already closed")
-		}
-
```

Comment:
This check does not make sense anymore since the `ioutils.NewReadCloserWrappe`  function  does check for double-close itself in v25.